src/main/java/com/google/devtools/build/lib/exec/AbstractSpawnStrategy.java

```diff
@@ -306,7 +306,18 @@ public ListenableFuture<Void> prefetchInputs() throws ForbiddenActionInputExcept
                         .setSpawn(FailureDetails.Spawn.newBuilder().setCode(code))
                         .build());
               } else {
-                throw e;
+                if (executionOptions.remoteRetryOnCacheError > 0) {
```

Comment:
Because this is an incompatible change, I would also like to guard it with `executionOptions.useNewExitCodeForLostInputs`. Then we have the same if condition, maybe change the code to:

```
if (executionOptions.useNewExitCodeForLostInputs || executionOptions.remoteRetryOnCacheError > 0) {
    if (BulkTransferException.allCausedByCacheNotFoundException(e)) {
        throw new EnvironmentalExecException(
                    e,
                    FailureDetail.newBuilder()
                        .setMessage("Failed to fetch blobs because they do not exist remotely.")
                        .setSpawn(FailureDetails.Spawn.newBuilder().setCode(Code.REMOTE_CACHE_EVICTED))
                        .build());
    } else {
        throw new EnvironmentalExecException(
                      e,
                      FailureDetail.newBuilder()
                          .setMessage("Failed to fetch blobs because from the remote cache.")
                          .setSpawn(
                              FailureDetails.Spawn.newBuilder().setCode(Code.REMOTE_CACHE_EVICTED))
                          .build());
    }
} else {
    throw e;
}
```

Reply:
That's not equivalent, and lacking the case where the legacy code path throws the `EnvironmentalExecException`.
I'll push a new version of the if though that has the experimental case and the legacy case clearly separated, so that deleting the legacy branch is very easy once cleaning up this experimental flag.